### PR TITLE
security bug: check at pub fn callback entry point

### PR DESCRIPTION
**Pull request by luciotato**
We need to check that `pub fn ft_resolve_transfer` is being called from this same contract.

### CODE REVIEW

File: near-contract-standards/src/fungible_token/implementation.rs [hunk 1]
```diff
@@ -224,6 +224,8 @@ impl FungibleTokenResolver for FungibleToken {
         receiver_id: ValidAccountId,
         amount: U128,
     ) -> U128 {
+        //check: MUST only be called as .then() callback from this contract
+        assert_eq!(env::predecessor_account_id(), env::current_account_id());
```
Comment (ilblackdragon) [inline near-contract-standards/src/fungible_token/implementation.rs:228]: there is actually `assert_self();` shortcut in near_sdk.

**Review by ilblackdragon — APPROVED** [review]

File: near-contract-standards/src/fungible_token/implementation.rs (hunk 1 above)
Comment (willemneal) [reply near-contract-standards/src/fungible_token/implementation.rs:228]: You can also use the `#[private]` macro above the method declaration, which adds this check.  Good catch @luciotato!

**Review by willemneal — APPROVED** [review]
LGTM!

File: near-contract-standards/src/fungible_token/implementation.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply near-contract-standards/src/fungible_token/implementation.rs:228]: Good catch. Let's switch to `#[private]` as @willemneal suggested

**Comment by luciotato** [comment]
tried #[private] but I get a compilation error. Maybe you can check that @willemneal

**Comment by luciotato** [comment]
Ok looks like #[private] is being added by the fungible_token macro implementation

The missing #[private] is in a piece of code that never gets used. The macro re-implements that code with #[private] (a few lines to call the internal resolver). 

Also you can't add #[private] to this superfluous code to avoid confusion, because #[private] depends on #[near_bindgen] and this internal `FungibleToken` struct MUST NOT be "bindgended". So the next best thing is to remove the code to avoid confusion

I'll drop this PR because in the current version there's no security bug.

**Pull request closed without merging**